frontend/packages/console-app/src/components/network-policies/network-policy-conditional-selector.tsx

```diff
@@ -4,7 +4,7 @@ import { PlusCircleIcon } from '@patternfly/react-icons';
 import { useTranslation } from 'react-i18next';
 import { AsyncComponent } from '@console/internal/components/utils/async';
 
-const NameValueEditorComponent = (props) => (
+const NameValueEditorComponent = (props: any) => (
```

Comment:
Small nit, doesn't mean a lot today but is future balanced if we go with the props of `AsyncComponent`. You can omit the static prop that is provided:

```suggestion
const NameValueEditorComponent = (props: Omit<AsyncComponentProps, 'loader'>) => (
```

Reply:
I agree. It is preferable to avoid type of `any`.

Reply:
I tried to fix that `any` before, but the really interesting type here is the one of `NameValueEditor` (which is JSX+proptypes, unfortunately it hasn't been typescriptified - I briefly tried to do that, but gave up as it was turning into a huge refactoring).

If you look at the `AsyncComponentProps`, it's defined as:
```
{
  loader: () => Promise<React.ComponentType>;
  LoadingComponent?: React.ReactNode;
} & any;
```
which is basically equal to `any` if we omit `loader`. So, yes I agree to do that change, unfortunately it's not particularly useful :(

Reply:
I agree it's not particularly useful today. But in an app this large, types can change and we may turn that into something else over time. In which case your component will still be typed as `any` despite having extra requirements. It's best we lean into types as we have them. Let the compiler help us do the checking when types change.